### PR TITLE
Push topN to Phoenix unconditionally

### LINKED ISSUE #7521: Push topN to Phoenix unconditionally when Phoenix 5.1.2 is released
Currently Phoenix overestimated the memory required to perform a topN query, so in #7490 the maximum limit in topN queries to be pushed down to Phoenix had to be limited (and thus Trino had a ensure the topN on its side again, even though Phoenix performed all the work already.)

When Phoenix 5.1.2 (4.16.1) is released, we can remove that limitation.

### PR DESCRIPTION
**Pull request by lhofhansl**
Fixes #7521

Only supports Phoenix 5, since I cannot update the Phoenix 4 client without breaking compatibility.

### CODE REVIEW

**Comment by lhofhansl** [comment]
Prerequisite for #8171

**Review by hashhar — APPROVED** [review]

File: plugin/trino-phoenix5/src/main/java/io/trino/plugin/phoenix5/PhoenixClient.java [hunk 1]
```diff
@@ -293,21 +292,14 @@ public boolean supportsTopN(ConnectorSession session, JdbcTableHandle handle, Li
...
-            return TopNFunction.sqlStandard(this::quoted).apply(query, sortItems, limit);
-        });
+        return Optional.of(TopNFunction.sqlStandard(this::quoted));
     }
 
     @Override
     public boolean isTopNGuaranteed(ConnectorSession session)
     {
+        // This remains false, because Phoenix only ensures that each
+        // split enforces its own limit.
```
Comment (findepi) [inline plugin/trino-phoenix5/src/main/java/io/trino/plugin/phoenix5/PhoenixClient.java:302]:
```
// There are multiple splits and TopN is not guaranteed across them.
```

Comment (lhofhansl) [reply plugin/trino-phoenix5/src/main/java/io/trino/plugin/phoenix5/PhoenixClient.java]: Done

**Comment by lhofhansl** [comment]
Thanks @hashhar

**Pull request merged**